plugin/environment.go

```diff
@@ -166,6 +167,17 @@ func (env *Environment) Activate(id string) (manifest *model.Manifest, activated
 		env.activePlugins.Store(pluginInfo.Manifest.Id, activePlugin)
 	}()
 
+	if pluginInfo.Manifest.MinServerVersion != "" {
+		minServerVersion, err := semver.Parse(pluginInfo.Manifest.MinServerVersion)
+		if err != nil {
+			return nil, false, fmt.Errorf("failed to parse MinServerVersion: %v", id)
+		}
+		serverVersion := semver.MustParse(model.CurrentVersion)
+		if serverVersion.LT(minServerVersion) {
```

Comment:
Does it return false when `serverVersion` is greater than the max value in `minServerVersion` (where `minServerVersion` is in range)?

I agree on required work to add test here in `plugin/environment.go` but would it make sense if we move some of the logic to `model/manifest.go` where we can test all cases there?  Usage could be like `pluginInfo.Manifest.MeetMinServerVersion()` or maybe pass the `model.CurrentVersion` as well so we don't need to mock it when testing.

Reply:
I moved the logic to `model/manifest.go` and added tests. Please review again.